android/src/main/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapper.kt

```diff
@@ -1,11 +1,36 @@
 package com.revenuecat.purchases.hybridcommon.mappers
 
 import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.models.PricingPhase
 import com.revenuecat.purchases.models.StoreProduct
 
+// TODOBC5: JOSH's temporary (???) extensions to support what was already there
+val StoreProduct.priceAmountMicros: Long
+    get() = this.oneTimeProductPrice?.priceAmountMicros ?: this.defaultOption?.pricingPhases?.lastOrNull()?.priceAmountMicros ?: 0
+val StoreProduct.price: String
+    get() = this.oneTimeProductPrice?.formattedPrice ?: this.defaultOption?.pricingPhases?.lastOrNull()?.formattedPrice ?: ""
+val StoreProduct.priceCurrencyCode: String
+    get() = this.oneTimeProductPrice?.currencyCode ?: this.defaultOption?.pricingPhases?.lastOrNull()?.priceCurrencyCode ?: ""
+val StoreProduct.freeTrialPeriod: String?
+    get() = this.defaultOption?.pricingPhases?.lastOrNull()?.billingPeriod
+
+private val StoreProduct.introductoryPhase: PricingPhase?
+    get() = this.defaultOption?.pricingPhases
+        ?.takeIf { it.size > 1 }
+        ?.firstOrNull()
+        ?.takeIf { it.priceAmountMicros > 0 }
+val StoreProduct.introductoryPrice: String?
+    get() = this.introductoryPhase?.formattedPrice
+val StoreProduct.introductoryPricePeriod: String?
+    get() = this.introductoryPhase?.billingPeriod
+val StoreProduct.introductoryPriceAmountMicros: Long
+    get() = this.introductoryPhase?.priceAmountMicros ?: 0
+val StoreProduct.introductoryPriceCycles: Int
+    get() = this.introductoryPhase?.billingCycleCount ?: 0
+
```

Comment:
No great way way for a developer to show like "3 free days, $3 off first week, then full price"

Reply:
this ^^ actually is possible now that we've confirmed you can only have 1 free trial and 1 intro price in an offer.

google COULD change things to allow for multiple intro price phases in one offer, and then this would not be comprehensive anymore (i.e. we'd need multiple introPrices)

BUT we are planning to expose SubscriptionOptions somewhere, so we could call out that if you want an offer with multiple intro phases, you should use that.